test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,769 @@
+        this.controllerWrapper.close();
+        this.serverListener.close();
+        this.serviceBuilder.close();
+        this.zkTestServer.close();
+        this.keyValueTableFactory.close();
+    }
+
+
+    /**
+     * Create Stream and KVTable with same name under same scope.
+     */
+    @Test
+    public void testCreateStreamKVTable() throws ExecutionException, InterruptedException {
+        String newScope = "NewScope";
+        KeyValueTableInfo newKVT = newKeyValueTableName(newScope);
+        String streamName = newKVT.getKeyValueTableName();
+        int streamSegments = 1;
+        String readerGroupName = "readerGroup";
+        String readerName = "reader";
+        String routingKey = "routingKey";
+        String streamEvent = "Stream event";
+        int numEvents = 100;
+        int writeCount = 0;
+        int readCount = 0;
+
+        final ScalingPolicy scalingPolicy = ScalingPolicy.fixed(streamSegments);
+        final StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(scalingPolicy)
+                .build();
+
+        assertTrue("The requested Scope must get created", this.controller.createScope(newScope).join());
+
+        assertTrue("The requested Stream must get created", this.controller.createStream(newScope,
+                streamName, config).join());
+
+        assertTrue("The requested KVT must get created",
+                this.controller.createKeyValueTable(newScope, newKVT.getKeyValueTableName(),
+                        DEFAULT_CONFIG).join());
+
+        assertEquals("Table partition value should be same as passed", DEFAULT_CONFIG.getPartitionCount(),
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegmentsForKeyValueTable(newScope, newKVT.getKeyValueTableName()),
+                        RuntimeException::new).getSegments().size());
+
+        assertEquals("The stream segment count should be same as passed", streamSegments,
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegments(newScope, streamName),
+                        RuntimeException::new).getSegments().size());
+
+        try (ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+             ClientFactoryImpl clientFactory = new ClientFactoryImpl(newScope, this.controller, connectionFactory);
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(newScope, this.controller, clientFactory)) {
+
+            log.info("Creating writer");
+            @Cleanup
+            final EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,
+                    new JavaSerializer<>(), EventWriterConfig.builder().build());
+
+            // start writing events to the stream.
+            for (int i = 0; i < numEvents; i++) {
+                log.info("Writing event {}", streamEvent);
+                writer.writeEvent(routingKey, streamEvent);
```

Comment:
Are you sure you don't want to wait on this operation to complete? Check the return value.